frontend/src/components/Gallery.tsx

```diff
@@ -5,31 +5,46 @@ import GalleryItems from "./GalleryItems";
 import "./GalleryItems.css"
 
 
-export default function Gallery(){
+export default function Gallery() {
 
 
-        const [countries, setCountries] = useState<Array<Country>>([])
+    const [countries, setCountries] = useState<Array<Country>>([])
 
     const [rangeValue, setRangeValue] = useState("3")
 
-    const names = countries.slice(0, parseInt(rangeValue)).map((c => <GalleryItems country={c}/>))
 
+    const radios = ["Africa", "Asia", "Europe", "Oceania", "North America", "South America"]
 
+    const [radio, setRadio] = useState("Africa");
 
+    const names = countries.filter((c) => c.continents.includes(radio)).slice(0, parseInt(rangeValue)).map((c =>
+        <GalleryItems country={c}/>))
 
-    useEffect( () => {
+
+    useEffect(() => {
 
         axios.get("https://restcountries.com/v3.1/all")
             .then(response => setCountries(response.data))
 
     }, [])
 
-    return(
-        <div className= "Gallery">
+    return (
+        <div className="Gallery">
             <h1>Countries</h1>
             <br/>
-                <input type= "range" min= "1" max = "250" defaultValue={rangeValue}
-                       onChange={(e) => setRangeValue(e.target.value)}/>
+            <input type="range" min="1" max="250" defaultValue={rangeValue}
+                   onChange={(e) => setRangeValue(e.target.value)}/>
+            <br/>
+            <br/>
+            {
+                radios.map((continents) => <div><input type="radio" id={continents} name="radioInput"
+                                                       defaultValue={radio}
+                                                       checked={radio === continents} onChange={(e) =>
+                    setRadio(e.target.id)}/>
+                    <label htmlFor={continents}>{continents}</label></div>)
```

Comment:
Here you are giving some of the RadioInputs an ID with a space. e.g. "North Amerika" 
Iam not sure if this is working like intended here, but i'am certain that when u want to adress this element in your css you will have some trouble. 

You could solve this by using this selector: 
```css
input[id='North Amerika']{
 
}
```
But i would suggest to just delete the Spaces from the String. 
Also your formating seems off.
```javascript
radios.map(continent=>{
   return (
      <div>
         <input type=radio id={continents.replaceAll(" ","")} name="radioInput" defaultValue={radio} checked={radio===continents} onChange={e=>setRadio(e.target.id)}/>
         <label htmlFor={continents.replaceAll(" ","")}>{continent}</label>
      </div>
   )
})
```